DailyKit/CallManager/CallClientSubjects.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+import Combine
+import Daily
+
+/// Subjects for ``CallClientProperty``s.
+final class CallClientSubjects {
+    let url: CurrentValueSubject<URL?, Never>
+    let callState: CurrentValueSubject<CallState, Never>
+    let camera: CurrentValueSubject<CallCamera, Never>
+    let microphone: CurrentValueSubject<CallMicrophone, Never>
+    let participants: CurrentValueSubject<CallParticipants, Never>
+
+    private var publishers: [AnyHashable: Any] = [:]
+
+    init(
+        url: URL?,
+        callState: CallState,
+        camera: CallCamera,
+        microphone: CallMicrophone,
+        participants: CallParticipants
+    ) {
+        self.url = CurrentValueSubject<URL?, Never>(url)
+        self.callState = CurrentValueSubject<CallState, Never>(callState)
+        self.camera = CurrentValueSubject<CallCamera, Never>(camera)
+        self.microphone = CurrentValueSubject<CallMicrophone, Never>(microphone)
+        self.participants = CurrentValueSubject<CallParticipants, Never>(participants)
+
+        setupPublishers()
+    }
+
+    private func setupPublishers() {
+        publishers[CallClientProperty.url] = url.eraseToAnyPublisher()
+        publishers[CallClientProperty.callState] = callState.eraseToAnyPublisher()
+        publishers[CallClientProperty.camera] = camera.eraseToAnyPublisher()
+        publishers[CallClientProperty.microphone] = microphone.eraseToAnyPublisher()
+        publishers[CallClientProperty.participants] = participants.eraseToAnyPublisher()
```

Comment:
```suggestion
        publishers[.url] = url.eraseToAnyPublisher()
        publishers[.callState] = callState.eraseToAnyPublisher()
        publishers[.camera] = camera.eraseToAnyPublisher()
        publishers[.microphone] = microphone.eraseToAnyPublisher()
        publishers[.participants] = participants.eraseToAnyPublisher()
```

Reply:
Unfortunately that doesn't work because the key for the dictionary is `AnyHashable`. With `CallClientProperty` being generic, I don't think there's a way to use it as a key directly. I'd be interested to know if there is though.

Reply:
Ah, gotcha!